pyproject.toml

```diff
@@ -2,5 +2,121 @@
 requires = ["setuptools>=64", "setuptools_scm[toml]>=6.2", "wheel"]
 build-backend = "setuptools.build_meta"
 
+[project]
+name = "python3-pip-skeleton"
+classifiers = [
+    "Development Status :: 3 - Alpha",
+    "License :: OSI Approved :: Apache Software License",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+]
+description = "One line description of your module"
+dependencies = [] # Add project dependencies here, e.g. ["click", "numpy"]
+dynamic = ["version"]
+license.file = "LICENSE"
+readme = "README.rst"
+requires-python = ">=3.8"
+
+[project.optional-dependencies]
+dev = [
+    "black",
+    "isort",
+    "mypy",
+    "flake8",
+    "flake8-isort",
+    "flake8-pyi",
+    "Flake8-pyproject",
+    "pipdeptree",
+    "pre-commit",
+    "pydata-sphinx-theme",
+    "pytest-cov",
+    "setuptools_scm[toml]>=6.2",
+    "sphinx-autobuild",
+    "sphinx-copybutton",
+    "sphinx-design",
+    "tox",
+    "tox-direct",
+    "types-mock",
+]
+
+[project.scripts]
+python3-pip-skeleton = "python3_pip_skeleton.__main__:main"
+
+[project.urls]
+GitHub = "https://github.com/DiamondLightSource/python3-pip-skeleton"
+
+[[project.authors]]
+email = "email@address.com"
+name = "Firstname Lastname"
+
+[[project.maintainers]]
+email = "email@address.com"
+name = "Firstname Lastname"
+
```

Comment:
I'd be tempted to omit this:
https://packaging.python.org/en/latest/specifications/core-metadata/#maintainer